context.go

```diff
@@ -2,28 +2,60 @@ package alog
 
 import "context"
 
-type key struct{}
+type stringKey struct{}
 
-var ctxkey = &key{}
+type structuredKey struct{}
+
+var stringCtxKey = &stringKey{}
+var structuredCtxKey = &structuredKey{}
```

Comment:
Hmm, let's make these two key-types non-pointers. It's weird and adds state in places none is necessary.